haystack/nodes/retriever/_embedding_encoder.py

```diff
@@ -374,9 +377,71 @@ def save(self, save_dir: Union[Path, str]):
         )
 
 
+class _OpenAIEmbeddingEncoder(_BaseEmbeddingEncoder):
+    def __init__(self, retriever: "EmbeddingRetriever"):
+        # pretrained embedding models coming from:
+        self.max_seq_len = retriever.max_seq_len
+        self.url = "https://api.openai.com/v1/embeddings"
+        self.api_key = retriever.api_key
+        model_class: str = next(
```

Comment:
Why not just `model_class = retriever.embedding_model`?

Reply:
Yeah, good point but I wanted to handle the case when users accidentally specify the full name of the model. Some might specify "ada", "babbage" etc and some might specify the full name. This way we handle properly both use cases.

Reply:
Makes sense, I'm just wondering, what if the user want to use `text-similarity-ada-001` model for example. In this case, we would silently use `text-search-ada-doc-001` / `text-search-ada-query-001` without the user knowing that.
We should also probably adapt the docstring of the param `embedding_model` of the `EmbeddingRetriever`, what do you think?

Reply:
Our use case is definitely [Text search embeddings](https://beta.openai.com/docs/guides/embeddings/text-search-embeddings)